java-checks/src/main/java/org/sonar/java/checks/KnownCapacityHashBasedCollectionCheck.java

```diff
@@ -69,4 +77,13 @@ private static String getIssueMessage(NewClassTree newClassTree) {
     return String.format("Replace this call to the constructor with the better suited static method %s", replacementMethod);
   }
 
+  private static List<JavaQuickFix> computeQuickFix(NewClassTree newClassTree) {
+    String replacementMethod = TYPES_TO_METHODS.get(newClassTree.symbolType().name()).replace("(int numMappings)", "");
+    JavaTextEdit edit = JavaTextEdit.replaceBetweenTree(newClassTree.firstToken(), newClassTree.identifier().lastToken(), replacementMethod);
+    JavaQuickFix quickFix = JavaQuickFix.newQuickFix("Replace with \"" + replacementMethod + "\".")
+      .addTextEdit(edit)
+      .build();
+    return List.of(quickFix);
+  }
```

Comment:
This can be simplified to return 1 quickfix

```suggestion
  private static JavaQuickFix computeQuickFix(NewClassTree newClassTree) {
    String replacementMethod = TYPES_TO_METHODS.get(newClassTree.symbolType().name()).replace("(int numMappings)", "");
    JavaTextEdit edit = JavaTextEdit.replaceBetweenTree(newClassTree.firstToken(), newClassTree.identifier().lastToken(), replacementMethod);
    return JavaQuickFix.newQuickFix("Replace with \"" + replacementMethod + "\".")
      .addTextEdit(edit)
      .build();
  }
```